src/main/java/randoop/resource/generator/MethodListGen.java

```diff
@@ -0,0 +1,252 @@
+                InputStream is = jar.getInputStream(jarEntry);
+                annotatedMethods.addAll(
+                    readClassFile(is, annotationCategory, qualifyingAnnotations));
+              }
+            } catch (IOException e) {
+              throw new RuntimeException("Failure to parse: " + e.getMessage());
+            }
+          });
+    } catch (IOException e) {
+      throw new RuntimeException("Failure to parse: " + e.getMessage());
+    }
+
+    // Sort in alphabetical order
+    Collections.sort(annotatedMethods);
+    return annotatedMethods;
+  }
+
+  /**
+   * Reads a class file using the given input stream and parses it for the desired annotations.
+   *
+   * @param classInputStream input stream used for reading
+   * @param annotationCategory type of annotation we are parsing for
+   * @param qualifyingAnnotations list of annotations
+   * @return list of methods with the desired annotations in fully qualified signature format
+   * @throws IOException if SceneLib fails to parse the class file
+   */
+  private static List<String> readClassFile(
+      InputStream classInputStream,
+      AnnotationCategory annotationCategory,
+      Collection<String> qualifyingAnnotations)
+      throws IOException {
+    List<String> annotatedMethods = new ArrayList<>();
+
+    // Invoke the parser for the specified class file
+    AScene scene = new AScene();
+    ClassFileReader.read(scene, classInputStream);
+
+    // Look through each class and its corresponding methods.
+    for (Map.Entry<String, AClass> entry : scene.classes.entrySet()) {
+      AClass aclass = entry.getValue();
+      for (Map.Entry<String, AMethod> m : aclass.methods.entrySet()) {
+        // Check annotations for the method
+        Collection<Annotation> annotationLocation =
+            annotationCategory.equals(AnnotationCategory.NON_DET)
+                ? m.getValue().returnType.tlAnnotationsHere
+                : m.getValue().tlAnnotationsHere;
+        for (Annotation a : annotationLocation) {
+          // Relocation changes the constant annotation comparisons.
+          // Thus, we need to append the 'randoop.' prefix to the annotation
+          // parsed from the class files.
+          if (qualifyingAnnotations.contains("randoop." + a.def.name.trim())) {
+            String fullyQualifiedName =
+                getFullyQualifiedSignatures(aclass, m.getValue().methodName);
+            annotatedMethods.add(fullyQualifiedName);
+            break;
+          }
+        }
+      }
+    }
+    return annotatedMethods;
+  }
+
+  /**
+   * Returns the fully qualified signature from the given class and signature in JVML format.
+   *
+   * @param aclass AClass that the method belongs to
+   * @param JVMLmethodSignature method signature in JVML
+   * @return fully qualified method signature
+   */
+  private static String getFullyQualifiedSignatures(AClass aclass, String JVMLmethodSignature) {
```

Comment:
The idiomatic way to capitalize the formal parameter name is `jvmlMethodSignature`.